sunbeam-python/sunbeam/core/checks.py

```diff
@@ -553,3 +555,63 @@ def run(self) -> bool:
         else:
             self.message = f"Juju controller {self.controller} is not registered."
             return False
+
+
+class LxdGroupCheck(Check):
+    """Check if user is member of lxd group."""
+
+    def __init__(self):
+        self.user = os.environ.get("USER")
+        self.group = "lxd"
+
+        super().__init__(
+            "Check for lxd group membership",
+            f"Checking if user {self.user} is member of group {self.group}",
+        )
+
+    def run(self) -> bool:
+        """Return false if user is not member of group.
+
+        Checks:
+            - User is part of group
+        """
+        if self.user not in grp.getgrnam(self.group).gr_mem:
```

Comment:
Would that return true if the user's in the LXD group, but current shell is not?

Reply:
Yes the current shell returns True irrespective of current group.
```
ubuntu@sunbeam1:~$ id
uid=1000(ubuntu) gid=584788(snap_daemon) groups=584788(snap_daemon),4(adm),20(dialout),24(cdrom),25(floppy),27(sudo),29(audio),30(dip),44(video),46(plugdev),119(netdev),120(lxd),1000(ubuntu)
ubuntu@sunbeam1:~$ 
ubuntu@sunbeam1:~$ python3
Python 3.10.12 (main, Sep 11 2024, 15:47:36) [GCC 11.4.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import grp
>>> grp.getgrnam("lxd").gr_mem
['ubuntu']
>>>
```